coalahtml/coala_html.py

```diff
@@ -29,13 +29,34 @@
 from coalib.output.printers.ListLogPrinter import ListLogPrinter
 
 
+def gen_file_set(dirpath):
+    """
+    Parse an output from `os.walk` and create a set of its contents.
+    :param dirpath:
+        Source directory.
+    :return:
+        A set of directory contents traversed recursively.
+    """
+    dir_content = os.walk(os.path.abspath(dirpath))
+    content_set = set()
+    for dir_tuple in dir_content:
+        root, dir_list, file_list = dir_tuple
+        for file_name in [root] + dir_list + file_list:
```

Comment:
I dislike this a little bit. You're creating a bit list, just for traversing it.
Doing stuff, like
```
dirpath = os.path.abspath(dirpath)
for root, dir_list, file_list in os.walk(dirpath):  # here I like to put os.walk here, as it will more apparent what you're doing, but not with abspath, cause it may get too complicated. Just personal preference, feel free to ignore.
        content_set.add(root)
        for file_name in itertools.chain(dir_list, file_list):  # this is not creating a huge list, so more effective on time and memory.
            content_set.add(file_name)
```
can be more effective.